src/styles/main.scss

```diff
@@ -1,3 +1,72 @@
+* {
+  box-sizing: border-box;
+}
+
+$days: (
+  "mon": 0,
+  "tue": 1,
+  "wed": 2,
+  "thu": 3,
+  "fri": 4,
+  "sat": 5,
+  "sun": 6
+);
+
+$day-size: 100px;
+$gap-size: 1px;
+$week-size: 7;
+$calendar-size: $week-size * ($day-size + $gap-size);
+
+html {
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
 body {
   margin: 0;
 }
+
+@for $i from 1 through 31 {
+  .calendar__card:nth-child(#{$i})::before {
+    content: "#{$i}";
+  }
+}
+
+@each $day, $value in $days {
+  .calendar_start-day--#{$day} :first-child {
+    margin-left: $value * ($day-size + $gap-size);
+  }
+}
+
+@for $i from 28 through 31 {
+  .calendar_month-length--#{$i} :nth-child(n + #{$i + 1}) {
+    display: none;
+  }
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  flex-grow: 2;
+  width: $calendar-size;
+  margin: 0 auto;
+}
+
+.calendar__card {
+  background-color: #eee;
+  width: $day-size;
+  height: $day-size;
+  border: solid 1px black;
+  padding: auto;
```

Comment:
1) You don't need padding here
2) `padding: auto` is not a thing, padding does not have an `auto` value. You can check that in the dev tools.